### DISCUSSION TITLE
Changing hover_language to fortran77

### DISCUSSION THREAD

**Discussion by Ester-topchik** [fortran-lang/fortls#318] [Q&A]
Hi. 
I get some "problems" from fortls:
1. "Unexpected end of scope at line " - at the first line of my .FOR file. 
2. Subroutine/Function definition before CONTAINS statement" - at the lines where subroutine definition begins. 
But If I build my project with GCC 7.2.0 compiler there are no problems, it works fine. I also get some warnings similar with those which I get during compilation. 
I suppose these 2 problems caused by wrong fortls configuration: "hover_language": " fortran90".
How can I change it to fortran77 if it's possible? Or maybe there are some other ways to solve these problems? 
P. S. I'm new here and sorry for bad English)

**Comment by gnikit** [comment]
Can you provide a minimal example to reproduce these errors

Don't expect the parser of `fortls` to be as robust or correct as that of a static compiler like GFortran, they are designed to perform different functions. `fortls` cannot compile your code.

`hover_language` is meant for the syntax highlighting features of code editors. `fortls` has some relatively robust heuristics to determine if a file is fixed or free form. Because these heuristics check entire files, any file that uses both fixed and free form in it will probably be parsed incorrectly.
